### PR TITLE
Replacing `User#warned` with `User#warned?`

### PR DESCRIPTION
**Pull request by jeremyf**
## What type of PR is this? (check all applicable)

- [x] Refactor

## Description

A simply refactor to introduce more idiomatic method names.

## Related Tickets & Documents

Related to #15624

## QA Instructions, Screenshots, Recordings

None.

### UI accessibility concerns?

None.

## Added/updated tests?

- [x] No, and this is why: refactor of method names

## [Forem core team only] How will this change be communicated?

- [x] This change does not need to be communicated, and this is why not: it's a simple refactor

### CODE REVIEW

**Review by Zhao-Andy — APPROVED** [review]

File: spec/system/admin/admin_bans_or_warns_user_spec.rb [hunk 1]
```diff
@@ -62,7 +62,7 @@ def unsuspend_user
 
     expect(user.suspended?).to eq(true)
     expect(user.trusted).to eq(false)
```
Comment (juliannatetreault) [inline spec/system/admin/admin_bans_or_warns_user_spec.rb:64]: With all of these refactors, are we also planning on changing `trusted` to `trusted?`

**Review by djuber — APPROVED** [review]
Makes sense to me.

**Review by juliannatetreault — APPROVED** [review]
Left a quick question, but otherwise looks good!

File: spec/system/admin/admin_bans_or_warns_user_spec.rb (hunk 1 above)
Comment (djuber) [reply spec/system/admin/admin_bans_or_warns_user_spec.rb:64]: maybe we already did (and it's just a change in the callers)?

https://github.com/forem/forem/blob/main/app/models/user.rb#L384

Comment (juliannatetreault) [reply spec/system/admin/admin_bans_or_warns_user_spec.rb:64]: I noticed that, @djuber 👀

**Pull request merged**